packages/protocol/genesis/GenerateGenesis.g.sol

```diff
@@ -77,70 +74,62 @@ contract TestGenerateGenesis is Test, AddressResolver {
+        assertEq(owner, addressManagerProxy.owner());
 
-        checkSavedAddress(addressManager, "SingletonTaikoL2Proxy", "taiko");
-        checkSavedAddress(
-            addressManager, "SingletonSignalServiceProxy", "signal_service"
-        );
+        checkSavedAddress(addressManagerProxy, "SingletonTaikoL2Proxy", "taiko");
+        checkSavedAddress(addressManagerProxy, "SingletonSignalServiceProxy", "signal_service");
 
-        TransparentUpgradeableProxy proxy = TransparentUpgradeableProxy(
-            payable(getPredeployedContractAddress("AddressManagerProxy"))
-        );
+        AddressManager addressManager =
+            AddressManager(getPredeployedContractAddress("AddressManager"));
 
         AddressManager newAddressManager = new AddressManager();
 
-        vm.startPrank(admin);
+        vm.startPrank(addressManager.owner());
 
-        proxy.upgradeTo(address(newAddressManager));
+        addressManager.upgradeTo(address(newAddressManager));
 
-        assertEq(proxy.implementation(), address(newAddressManager));
         vm.stopPrank();
     }
 
     function testSingletonAddressManagerForSingletons() public {
-        AddressManager addressManager =
-            AddressManager(getPredeployedContractAddress("SingletonAddressManagerForSingletonsProxy"));
-
-        assertEq(owner, addressManager.owner());
-
-        checkSavedAddress(addressManager, "SingletonBridgeProxy", "bridge");
-        checkSavedAddress(addressManager, "SingletonERC20VaultProxy", "erc20_vault");
-        checkSavedAddress(addressManager, "SingletonERC721VaultProxy", "erc721_vault");
-        checkSavedAddress(addressManager, "SingletonERC1155VaultProxy", "erc1155_vault");
-        checkSavedAddress(
-            addressManager, "SingletonSignalServiceProxy", "signal_service"
+        AddressManager addressManagerProxy = AddressManager(
+            getPredeployedContractAddress("SingletonAddressManagerForSingletonsProxy")
         );
 
-        TransparentUpgradeableProxy proxy = TransparentUpgradeableProxy(
-            payable(getPredeployedContractAddress("SingletonAddressManagerForSingletonsProxy"))
-        );
+        assertEq(owner, addressManagerProxy.owner());
 
+        checkSavedAddress(addressManagerProxy, "SingletonBridgeProxy", "bridge");
+        checkSavedAddress(addressManagerProxy, "SingletonERC20VaultProxy", "erc20_vault");
+        checkSavedAddress(addressManagerProxy, "SingletonERC721VaultProxy", "erc721_vault");
+        checkSavedAddress(addressManagerProxy, "SingletonERC1155VaultProxy", "erc1155_vault");
+        checkSavedAddress(addressManagerProxy, "SingletonSignalServiceProxy", "signal_service");
+
+        AddressManager addressManager =
+            AddressManager(getPredeployedContractAddress("AddressManagerForSingletons"));
         AddressManager newAddressManager = new AddressManager();
 
-        vm.startPrank(admin);
+        vm.startPrank(addressManager.owner());
```

Comment:
For all L2 contracts on mainnet, the owner shall be our security council.

    address public constant MAINNET_SECURITY_COUNCIL = 0x7C50d60743D3FCe5a39FdbF687AFbAe5acFF49Fd;